### PR TITLE
#254 Container Service Managed Clusters Agent Pools support

### LINKED ISSUE #254: Container Service Managed Clusters Agent Pools support
As an engineer, I would like to create a function/cmdlet to support Container Service Managed Clusters Agent Pools so that we can support future customer's needs.

The bicep resource is `Microsoft.ContainerService/managedClusters/agentPools`

- [ ] Create `Confirm-ContainerServiceManagedClusterAgentPool.ps1` and unit tests
- [ ] Add `ContainerServiceManagedClusterAgentPool` to `ResourceType` enum
- [ ] Add `ContainerServiceManagedClusterAgentPool.cs` to `BenchPress\Generators\ResourceTypes` folder
- [ ] Wire up `Confirm-ContainerServiceManagedClusterAgentPool` to `Get-ResourceByType.ps1`
- [ ] Create example tests, bicep and documentation in `examples` folder

### PR DESCRIPTION
**Pull request by marclerwick**
Closes #254

### CODE REVIEW

File: Modules/BenchPress.Azure/Public/Confirm-AksNodePool.ps1 [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+function Confirm-AksNodePool {
+  <#
+    .SYNOPSIS
+      Confirms that an AKS Node Pool exists.
+
+    .DESCRIPTION
+      The Confirm-AzBPAksNodePool cmdlet gets an AKS Node Pool using the specified AKS Node Pool and Resource Group
+      name.
+
+    .PARAMETER AKSName
```
Comment (msha1026) [inline Modules/BenchPress.Azure/Public/Confirm-AksNodePool.ps1:16]: Doc parameter list and actual parameter list do not match.

File: Modules/BenchPress.Azure/Public/Confirm-AksNodePool.ps1 [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      The name of the AKS Node Pool
+
+    .PARAMETER ResourceGroupName
+      The name of the Resource Group
+
+    .EXAMPLE
+      Confirm-AzBPAksNodePool -AKSName "benchpresstest" -ResourceGroupName "rgbenchpresstest"
```
Comment (msha1026) [inline Modules/BenchPress.Azure/Public/Confirm-AksNodePool.ps1:23]:
```suggestion
      Confirm-AzBPAksNodePool -Name "benchpresstest" -ClusterName "testcluster" -ResourceGroupName "rgbenchpresstest"
```

File: Modules/BenchPress.Azure/Public/Confirm-AksNodePool.ps1 [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    [Parameter(Mandatory=$true)]
+    [string]$Name
+  )
+  Begin {
+    $ConnectResults = Connect-Account
+  }
+  Process {
+    $Resource = Get-AzAksNodePool -ResourceGroupName $ResourceGroupName -ClusterName $ClusterName -Name $Name
+
+    [ConfirmResult]::new($Resource, $ConnectResults.AuthenticationData)
```
Comment (msha1026) [inline Modules/BenchPress.Azure/Public/Confirm-AksNodePool.ps1:49]:
```suggestion
    $connectResults = Connect-Account
  }
  Process {
    $resource = Get-AzAksNodePool -ResourceGroupName $ResourceGroupName -ClusterName $ClusterName -Name $Name

    [ConfirmResult]::new($resource, $connectResults.AuthenticationData)
```

File: Modules/BenchPress.Azure/Public/Get-ResourceByType.ps1 [hunk 4]
```diff
@@ -127,6 +134,14 @@ function Get-ResourceByType {
       "AksCluster" {
         return Confirm-AksCluster -AKSName $ResourceName -ResourceGroupName $ResourceGroupName
       }
+      "AksNodePool" {
+        $params = @{
+          ResourceGroupName = $ResourceGroupName
+          ClusterName = $ClusterName
+          Name = $ResourceName
```
Comment (msha1026) [inline Modules/BenchPress.Azure/Public/Get-ResourceByType.ps1:141]: Edit the block so that the `=` are aligned

**Review by msha1026 — APPROVED** [review]
Other than small fixes, looks good!

File: Modules/BenchPress.Azure/Public/Confirm-AksNodePool.ps1 (hunk 3 above)
Comment (marclerwick) [reply Modules/BenchPress.Azure/Public/Confirm-AksNodePool.ps1:49]: haha, I had already done this based on the other review :-D

File: Modules/BenchPress.Azure/Public/Get-ResourceByType.ps1 (hunk 4 above)
Comment (marclerwick) [reply Modules/BenchPress.Azure/Public/Get-ResourceByType.ps1]: Done!

File: Modules/BenchPress.Azure/Public/Confirm-AksNodePool.ps1 (hunk 2 above)
Comment (marclerwick) [reply Modules/BenchPress.Azure/Public/Confirm-AksNodePool.ps1]: Done!

File: Modules/BenchPress.Azure/Public/Confirm-AksNodePool.ps1 (hunk 1 above)
Comment (marclerwick) [reply Modules/BenchPress.Azure/Public/Confirm-AksNodePool.ps1]: Done!

File: examples/AksCluster/AKSCluster.Tests.ps1 [hunk 5]
```diff
@@ -87,6 +87,66 @@ Describe 'Verify AKS Cluster' {
...
+      ClusterName = $aksName
+      Name = $noNodePoolName
+      ErrorAction = "SilentlyContinue"
+    }
+
+    #act and assert
+    (Confirm-AzBPAksNodePool @params).Success | Should -Be $false
+  }
+
+  It "Should contain an AKS Node Pool named $aksName" {
```
Comment (ozboms) [inline examples/AksCluster/AKSCluster.Tests.ps1:145]: Could you delete the `BeDeployed `test?

File: examples/AksCluster/AKSCluster.Tests.ps1 [hunk 6]
```diff
@@ -87,6 +87,66 @@ Describe 'Verify AKS Cluster' {
...
+      ResourceName      = $nodePoolName
+      PropertyKey       = "Name"
+      PropertyValue     = $nodePoolName
+    }
+
+    #act and assert
+    (Confirm-AzBPResource @params).Success | Should -Be $true
+  }
+
+  It "Should contain an AKS Node Pool with the name $aksNodePoolName" {
```
Comment (ozboms) [inline examples/AksCluster/AKSCluster.Tests.ps1:124]:
```suggestion
  It "Should contain an AKS Node Pool with the name $nodePoolName" {
```

Comment (marclerwick) [reply examples/AksCluster/AKSCluster.Tests.ps1]: Done!

File: examples/AksCluster/AKSCluster.Tests.ps1 (hunk 5 above)
Comment (marclerwick) [reply examples/AksCluster/AKSCluster.Tests.ps1]: Done with the merge from main.

**Review by ozboms — APPROVED** [review]

File: Modules/BenchPress.Azure/Public/Confirm-AksNodePool.ps1 [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+. $PSScriptRoot/../Private/Connect-Account.ps1
+# end INLINE_SKIP
+
+function Confirm-AksNodePool {
+  <#
+    .SYNOPSIS
+      Confirms that an AKS Node Pool exists.
+
+    .DESCRIPTION
+      The Confirm-AzBPAksNodePool cmdlet gets an AKS Node Pool using the specified AKS Node Pool and Resource Group
```
Comment (msha1026) [inline Modules/BenchPress.Azure/Public/Confirm-AksNodePool.ps1:13]:
```suggestion
      The Confirm-AzBPAksNodePool cmdlet gets an AKS Node Pool using the specified AKS Node Pool and Resource Group names.
```

File: Modules/BenchPress.Azure/Public/Confirm-AksNodePool.ps1 [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+    .PARAMETER ResourceGroupName
+      The name of the Resource Group.
+
+    .PARAMETER ClusterName
+      The name of the managed cluster resource.
```
Comment (msha1026) [inline Modules/BenchPress.Azure/Public/Confirm-AksNodePool.ps1:19]:
```suggestion
      The name of the Managed Cluster.
```

File: Modules/BenchPress.Azure/Public/Confirm-AksNodePool.ps1 [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+
+    .PARAMETER Name
+      The name of the Node Pool
```
Comment (msha1026) [inline Modules/BenchPress.Azure/Public/Confirm-AksNodePool.ps1:22]:
```suggestion
      The name of the Node Pool.
```

File: Modules/BenchPress.Azure/Public/Confirm-Resource.ps1 [hunk 10]
```diff
@@ -68,6 +68,10 @@ function Confirm-Resource {
       If testing an Azure resource that is associated with a Job (e.g., Stream Analytics Output), the name of
       the associated Job.
 
+    .PARAMETER ClusterName
+      If the Azure resource is associated with an AKS Cluster (e.g, AKS Node Pool) this is the parameter to use to pass
+      the AKS cluster name.
```
Comment (msha1026) [inline Modules/BenchPress.Azure/Public/Confirm-Resource.ps1:73]:
```suggestion
      If testing an Azure resource that is associated with an AKS Cluster (e.g, AKS Node Pool), the name of
      the associated AKS Cluster.
```

File: Modules/BenchPress.Azure/Public/Get-ResourceByType.ps1 [hunk 11]
```diff
@@ -87,6 +88,10 @@ function Get-ResourceByType {
       If testing an Azure resource that is associated with a Job (e.g., Stream Analytics Output), the name of
       the associated Job.
 
+    .PARAMETER ClusterName
+      If the Azure resource is associated with an AKS Cluster (e.g, AKS Node Pool) this is the parameter to use to pass
+      the AKS cluster name.
```
Comment (msha1026) [inline Modules/BenchPress.Azure/Public/Get-ResourceByType.ps1:93]:
```suggestion
      If testing an Azure resource that is associated with an AKS Cluster (e.g, AKS Node Pool), the name of
      the associated AKS Cluster.
```

File: examples/AksCluster/AKSCluster.Tests.ps1 [hunk 12]
```diff
@@ -57,6 +57,61 @@ Describe 'Verify AKS Cluster' {
...
+      ResourceName      = $nodePoolName
+      PropertyKey       = "Name"
+      PropertyValue     = $nodePoolName
+    }
+
+    # act and assert
+    Confirm-AzBPResource @params | Should -BeSuccessful
+  }
+
+  It "Should contain an AKS Node Pool with the name $nodePoolName" {
```
Comment (msha1026) [inline examples/AksCluster/AKSCluster.Tests.ps1:94]:
```suggestion
  It "Should contain an AKS Node Pool named $nodePoolName" {
```

File: examples/AksCluster/AKSCluster.Tests.ps1 [hunk 13]
```diff
@@ -57,6 +57,61 @@ Describe 'Verify AKS Cluster' {
...
+    Confirm-AzBPAksNodePool -ResourceGroupName $rgName -ClusterName $aksName -Name $nodePoolName | Should -BeSuccessful
+  }
+
+  It "Should not contain an AKS Node Pool with the name $noNodePoolName" {
```
Comment (msha1026) [inline examples/AksCluster/AKSCluster.Tests.ps1:98]:
```suggestion
  It "Should not contain an AKS Node Pool named $noNodePoolName" {
```

File: Modules/BenchPress.Azure/Public/Confirm-AksNodePool.ps1 [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+function Confirm-AksNodePool {
+  <#
+    .SYNOPSIS
+      Confirms that an AKS Node Pool exists.
+
+    .DESCRIPTION
+      The Confirm-AzBPAksNodePool cmdlet gets an AKS Node Pool using the specified AKS Node Pool and Resource Group names.
+
+    .PARAMETER ResourceGroupName
+      The name of the Resource Group.
```
Comment (msha1026) [inline Modules/BenchPress.Azure/Public/Confirm-AksNodePool.ps1:16]:
```suggestion
      The name of the Resource Group. The name is case insensitive.
```

File: Modules/BenchPress.Azure/Public/Confirm-Resource.ps1 (hunk 10 above)
Comment (marclerwick) [reply Modules/BenchPress.Azure/Public/Confirm-Resource.ps1:73]: Fixed

File: Modules/BenchPress.Azure/Public/Get-ResourceByType.ps1 (hunk 11 above)
Comment (marclerwick) [reply Modules/BenchPress.Azure/Public/Get-ResourceByType.ps1:93]: Fixed.

**Review by msha1026 — APPROVED** [review]

**Pull request merged**